backend/src/database/libraries.go

```diff
@@ -44,15 +44,15 @@ func (db *DB) GetLibraryByID(id int) (*models.Library, error) {
 	return &library, nil
 }
 
-func (db *DB) ToggleLibraryVisibility(id int) error {
+func (db *DB) ToggleLibraryVisibility(id int) (models.Library, error) {
```

Comment:
can we return by reference here?

Reply:
made change to return by reference, and renamed to ToggleVisibilityAndRetrieveLibrary because the library is needed for  bucket logic.